### PR TITLE
Added documentation for the locale in the URL

### PR DESCRIPTION
**Pull request by robfe11**
## Overview

Updated the URI structure documentation with guidelines for locale.

### Features

- added Locale section
- added link to F5 diagram

#### Meta

- [X] provide a descriptive topic
- [X] provide an overview of contribution
- [X] no sensitive content included, such as:
  - security & privacy policy violating content
  - content considered competitive intelligence
  - keys, tokens or credentials
- [X] documentation format follows [this template][template]
- [X] fork is up to date ([see this guide from github](https://help.github.com/articles/syncing-a-fork/))
- [X] "work in progress" commits are squashed ([see "Squashing Commits"](https://git-scm.com/book/id/v2/Git-Tools-Rewriting-History))
- [ ] commits follow the [Karma][karma-format] or [Commitizen](https://www.npmjs.com/package/commitizen) format

[template]: ../.template.md
[karma-format]: https://karma-runner.github.io/1.0/dev/git-commit-msg.html

### CODE REVIEW

File: development/uri-structure.md [hunk 1]
```diff
@@ -12,23 +12,55 @@ The purpose of this document is to drive consistency for structuring our urls an
...
-What belongs in `[/path]` portion of URI?
+What belongs in `[/locale][/path]` portion of URI?
 
 ### Routing
 
-Once specific context paths are created in BTO F5 to route to TELUS digital inbound proxies, from there we have control to route to our different applications.
+Once specific context paths are created in BTO F5 to route to TELUS digital inbound proxies, from there we have control to route to our different applications.  Here is a diagram describing the routing for a telus.com request:
 
-#### www.wcstage.telus.com
+https://docs.google.com/drawings/d/1yUxOCdKRciYD7TvY_IXwzO2zW2G3ka4cdcX8SJfhSDA/edit
```
Comment (ahmadnassri) [inline development/uri-structure.md]: use markdown link syntax on the word "here" or "diagram"

File: development/uri-structure.md [hunk 2]
```diff
@@ -12,23 +12,55 @@ The purpose of this document is to drive consistency for structuring our urls an
...
+
+### Locale
+
+The `[/locale]` segment describes what locale the content is intended for. The locale can contain the language and region.  Language is mandatory, and region is optional.  Example URL:
+
+`www.telus.com/en/bc/internet`
+`www.telus.com/en/business` 
+
+As of September 2017, the `[/locale]` is set through cookies (Java & PHP stack) or URL segments (RA).  On the Java & PHP stack, the cookies will overwrite the locale that is provided through the URL.
```
Comment (ahmadnassri) [inline development/uri-structure.md]: lets only document the desired standard here, not the legacy.

File: development/uri-structure.md [hunk 3]
```diff
@@ -12,23 +12,55 @@ The purpose of this document is to drive consistency for structuring our urls an
...
+
+### Locale
+
+The `[/locale]` segment describes what locale the content is intended for. The locale can contain the language and region.  Language is mandatory, and region is optional.  Example URL:
+
+`www.telus.com/en/bc/internet`
+`www.telus.com/en/business` 
+
+As of September 2017, the `[/locale]` is set through cookies (Java & PHP stack) or URL segments (RA).  On the Java & PHP stack, the cookies will overwrite the locale that is provided through the URL.  
+The intendend behaviour going forward is to **not use cookies** to set the locale, as the applications locale should be set through the `[/locale]` URL segments.
```
Comment (ahmadnassri) [inline development/uri-structure.md]: you can reference "historically using cookies" here and declare that non-desirable

File: development/uri-structure.md [hunk 4]
```diff
@@ -12,23 +12,55 @@ The purpose of this document is to drive consistency for structuring our urls an
...
+
+`www.telus.com/en/bc/internet`
+`www.telus.com/en/business` 
+
+As of September 2017, the `[/locale]` is set through cookies (Java & PHP stack) or URL segments (RA).  On the Java & PHP stack, the cookies will overwrite the locale that is provided through the URL.  
+The intendend behaviour going forward is to **not use cookies** to set the locale, as the applications locale should be set through the `[/locale]` URL segments. 
+
+#### Available Locales
+
+Currently, only `en` and `fr` are supported by the [F5 routing][f5-www.telus.com].
```
Comment (ahmadnassri) [inline development/uri-structure.md]: is this url okay to share publically?

File: development/uri-structure.md [hunk 5]
```diff
@@ -12,23 +12,55 @@ The purpose of this document is to drive consistency for structuring our urls an
...
+
+For the regions, the following URI segments are supported: 
+* British Columbia - `bc`
```
Comment (ahmadnassri) [inline development/uri-structure.md]: make a table here instead? might look visually better

File: development/uri-structure.md (hunk 1 above)
Comment (githubjosh) [reply development/uri-structure.md]: Should we be referencing assets in Google docs? I thought you needed to be manually added to view them. Isn't that partly why we moved to Confluence? Curious

File: development/uri-structure.md (hunk 2 above)
Comment (githubjosh) [reply development/uri-structure.md]: It is helpful to understand though

File: development/uri-structure.md (hunk 3 above)
Comment (githubjosh) [reply development/uri-structure.md]: Amen

File: development/uri-structure.md (hunk 1 above)
Comment (robfe11) [reply development/uri-structure.md]: Anyone with a telus account can view it.  It's a living diagram, that's why I linked to it.

File: development/uri-structure.md (hunk 4 above)
Comment (robfe11) [reply development/uri-structure.md]: It's within our private github, so I assumed it wouldn't be an issue.

Comment (githubjosh) [reply development/uri-structure.md]: @ahmadnassri any guidance where these docs should live?  Confluence,  other?

File: development/uri-structure.md [hunk 6]
```diff
@@ -12,23 +12,56 @@ The purpose of this document is to drive consistency for structuring our urls an
...
 
 ## How
 
-What belongs in `[/path]` portion of URI?
+What belongs in `[/locale][/path]` portion of URI?
 
 ### Routing
 
-Once specific context paths are created in BTO F5 to route to TELUS digital inbound proxies, from there we have control to route to our different applications.
+Once specific context paths are created in BTO F5 to route to TELUS digital inbound proxies, from there we have control to route to our different applications.  [Here is a diagram](https://docs.google.com/drawings/d/1yUxOCdKRciYD7TvY_IXwzO2zW2G3ka4cdcX8SJfhSDA/edit) describing the routing for a telus.com request.
```
Comment (ahmadnassri) [inline development/uri-structure.md]: @robfe11 can you bring this diagram into the repo plz?

maybe under `/development/_assets`

Comment (crimeminister) [reply development/uri-structure.md]: I'll make the addition of the linked image in a follow-up PR, assuming that this PR is mergeable. @githubjosh I have heard you express interest in our path structure with an eye to SEO. Mind weighing in on this change?

**Comment by robfe11** [comment]
@crimeminister Thanks for adding the SVG, I made a small edit to add the doc URL below the SVG in case it needs to be updated in the future.  

@ahmadnassri This PR should be ready to be merged now.

**Pull request merged**